### PR TITLE
Add global attributes to IODA conversion output files

### LINKED ISSUE #685: Add global attributes to IODA output
Might include map of mission-to-index-numbers for RADS

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
- Adds facility for global attributes to IODA converters
- Adds input observation source files as global attribute to all converts
- Adds to the rads converter attribute with mission-to-mission-number-index (to decode mission index associated with individual observations
- Adds lit reference to rads converter (to show off reading from attributes in originating file)

In the process of this the function in the rads converter that assigned the mission numbers was removed and the map declaration and assignment moved to the converter method to allow access to the map to convert to a string. Am open to other ways to approach this.

Resolves https://github.com/NOAA-EMC/GDASApp/issues/685

### CODE REVIEW

File: utils/obsproc/Rads2Ioda.h [hunk 1]
```diff
@@ -57,10 +57,31 @@ namespace gdasapp {
       ncFile.getVar("time_mjd").getVar(datetime);
       iodaVars.referenceDate_ = "seconds since 1858-11-17T00:00:00Z";
 
+      std::map<std::string, int> altimeterMap;
```
Comment (guillaumevernieres) [inline utils/obsproc/Rads2Ioda.h:60]: why is this not a function anymore

**Review by guillaumevernieres — COMMENTED** [review]
does it concatenate the global attributes properly? Can you paste an ncdump of the output ioda file when processing 2+ RADS files?

**Comment by guillaumevernieres** [comment]
@AndrewEichmann-NOAA , ~side comment on you PR description: "resolves" is not a github keyword, and your issue will not close when the PR is merged.~

File: utils/obsproc/Rads2Ioda.h (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply utils/obsproc/Rads2Ioda.h:60]: So the map can be converted to a string that can be assigned to a IodaVar member

Comment (AndrewEichmann-NOAA) [reply utils/obsproc/Rads2Ioda.h:60]: It winds up taking less lines just to declare, assign, and use the map in-line

**Review by apchoiCMD — APPROVED** [review]

**Comment by CoryMartin-NOAA** [comment]
> @AndrewEichmann-NOAA , side comment on you PR description: "resolves" is not a github keyword, and your issue will not close when the PR is merged.

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests is the list, resolves is there...

**Review by guillaumevernieres — APPROVED** [review]
Nice @AndrewEichmann-NOAA :
```
// global attributes:
		string :_ioda_layout = "ObsGroup" ;
		:_ioda_layout_version = 0 ;
		string :obs_source_files = "rads_adt_3b_2021181.nc4" ;
		string :mission_index = "CRYOSAT2 = 6 JASON-1 = 3 JASON-2 = 4 JASON-3 = 5 SARAL = 7 SNTNL-3A = 1 SNTNL-3B = 2 " ;
		string :references = "RADS Data Manual, Version 4.2 or later" ;
```

**Pull request merged**